Userland/Libraries/LibWeb/Cookie/ParsedCookie.cpp

```diff
@@ -140,7 +141,7 @@ void process_attribute(ParsedCookie& parsed_cookie, StringView attribute_name, S
     } else if (attribute_name.equals_ignoring_ascii_case("Domain"sv)) {
         on_domain_attribute(parsed_cookie, attribute_value);
     } else if (attribute_name.equals_ignoring_ascii_case("Path"sv)) {
-        on_path_attribute(parsed_cookie, attribute_value);
+        on_path_attribute(parsed_cookie, MUST(String::from_utf8(attribute_value)));
```

Comment:
Seems a little weird to convert this to a string here instead of either when `unparsed_attributes` is created above or when `parsed_cookie.path` is set below

Reply:
Right, it definitely does look a bit weird because of the asymmetry, and I was trying to think about best place to put it. My logic here was:

 * Doing it higher would result in making a String in some cases where we only need a StringView
 * Putting it into this function doesn't make it obvious that this function needs to take a copy of a String

I eventually convinced myself to put it here, but can be convinced otherwise

Reply:
> Putting it into this function doesn't make it obvious that this function needs to take a copy of a String

I don't think that's particularly a concern :man_shrugging:

Reply:
I don't think I was very clear on that point. I'm meaning that I don't think it's very nice to have a function call take a `StringView` only to call `String::from_utf8` on it for the same reason that we shouldn't be returning a `StringView` on a function which has a `String` instead of a `String const&` / `String`. If we already have a `String`, we don't want to be calling `from_utf8` on it later on.

Reply:
I don't follow this point here -

> I don't think it's very nice to have a function call take a StringView only to call String::from_utf8 on it

But that's exactly what you have `parse_attributes` doing in this patch anyways? Doing it inside `on_path_attribute` instead does not make that worse, and looks more consistent in this case (and avoids an allocation if the first char isn't `/`, but I'm sure that's not impactful).

Reply:
In most cases a String isn't needed, it's only `on_path_attribute` which needs one. In general, I've been trying to put any construction of a `String` as high up in call stack as possible since there could be another function in that chain which already has a String.

But that's a good point about the `/` case, because I could use same logic as there (although I was originally considering that as bad-path case).

Anyway, I'll make that change, thanks :^)